jumanji/environments/routing/multi_cvrp/generator.py

```diff
@@ -0,0 +1,144 @@
+        ) = get_init_settings(self.num_customers, self.num_vehicles)
+
+        self._max_end_window = self._max_start_window + self._time_window_length
+
+    @property
+    def num_customers(self) -> int:
+        return self._num_customers
+
+    @property
+    def num_vehicles(self) -> int:
+        return self._num_vehicles
+
+    @abc.abstractmethod
+    def __call__(self, key: chex.PRNGKey) -> State:
+        """Generates a `MultiCVRP` state.
+
+        Returns:
+            A `MultiCVRP` state.
+        """
+
+
+class UniformRandomGenerator(Generator):
+    """Randomly (uniformly) places the customers and vehicles on a fixed size map."""
+
+    def __init__(self, num_customers: int, num_vehicles: int) -> None:
+        """Initialises a MultiCVRP generator, used to generate the problem.
+
+        Args:
+            num_customers: number of customer nodes in the environment.
+            num_vehicles: number of vehicles in the environment.
+        """
+        super().__init__(num_customers, num_vehicles)
+
+    def __call__(self, key: chex.PRNGKey) -> State:
+        """Generates a `MACVPR` state.
+
+        Returns:
+            A `MACVPR` state.
+        """
+
+        # This split is uncessary, but it makes the code more readable.
+        problem_key, _ = jax.random.split(key)
+
+        total_capacity = self._max_capacity * self._num_vehicles
+
+        (
+            node_coordinates,
+            node_demands,
+            window_start_times,
+            window_end_times,
+            early_coefs,
+            late_coefs,
+        ) = generate_uniform_random_problem(
+            problem_key,
+            self._num_customers,
+            total_capacity,
+            self._map_max,
+            self._customer_demand_max,
+            self._max_start_window,
+            self._time_window_length,
+            self._early_coef_rand,
+            self._late_coef_rand,
+        )
+        capacities = jnp.ones(self._num_vehicles, dtype=jnp.int16) * self._max_capacity
+        state = State(
+            nodes=Node(coordinates=node_coordinates, demands=node_demands),
+            windows=TimeWindow(start=window_start_times, end=window_end_times),
+            coeffs=PenalityCoeff(early=early_coefs, late=late_coefs),
+            vehicles=StateVehicle(
+                positions=jnp.int16([DEPOT_IDX] * self._num_vehicles),
+                local_times=jnp.zeros(self._num_vehicles, dtype=jnp.float32),
+                capacities=capacities,
+                distances=jnp.zeros(self._num_vehicles, dtype=jnp.float32),
+                time_penalties=jnp.zeros(self._num_vehicles, dtype=jnp.float32),
+            ),
+            order=jnp.zeros(
+                (self._num_vehicles, 2 * self._num_customers), dtype=jnp.int16
+            ),
+            action_mask=create_action_mask(node_demands, capacities),
+            step_count=jnp.ones((), dtype=jnp.int16),
+            key=jax.random.PRNGKey(0),
```

Comment:
The state key is used for auto reset. It must not be set to 0 otherwise all states will always be reset in an identical way. This probably makes training very wrong and unstable. It is worth retraining.
```suggestion
            key=key,
```

Reply:
I see. Thanks, I did not know that. Update :+1: